File: modules/jooby-openapi/src/test/java/issues/i1905/App1905.java
package issues.i1905;

import io.jooby.Jooby;

public class App1905 extends Jooby {
  {
    install(SubApp1905::new);

    install("/inline", SubApp1905::new);

    install("/lambda-body", () -> {
      SubApp1905 app = new SubApp1905();
      doSomething(app);
      return app;
    });

    install("/instance-reference", this::instanceMethod);

    install("/static-reference", App1905::staticMethod);
  }

  private static SubApp1905 staticMethod() {
    SubApp1905 app = new SubApp1905();
    System.out.println(app);
    return app;
  }

  private SubApp1905 instanceMethod() {
    SubApp1905 app = new SubApp1905();
    System.out.println(app);
    return app;
  }

  private void doSomething(SubApp1905 app) {

  }
}

File: modules/jooby-openapi/src/test/kotlin/kt/i1905/App1905.kt
/*
 * Jooby https://jooby.io
 * Apache License Version 2.0 https://jooby.io/LICENSE.txt
 * Copyright 2014 Edgar Espina
 */
package kt.i1905

import io.jooby.kt.Kooby

fun staticCreateApp(): SubApp1905 {
  return SubApp1905()
}

class App1905 :
  Kooby({
    fun instanceCreateApp(): SubApp1905 {
      return SubApp1905()
    }

    install(::SubApp1905)

    install("/static/ref", ::staticCreateApp)

    install("/instance/ref", ::instanceCreateApp)

    install("/supplier") {
      val app = SubApp1905()
      print(app)
      app
    }
  })
